staging/src/k8s.io/client-go/transport/token_source.go

```diff
@@ -72,14 +92,20 @@ func NewCachedTokenSource(ts oauth2.TokenSource) oauth2.TokenSource {
 type tokenSourceTransport struct {
 	base http.RoundTripper
 	ort  http.RoundTripper
+	src  ResettableTokenSource
 }
 
 func (tst *tokenSourceTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	// This is to allow --token to override other bearer token providers.
 	if req.Header.Get("Authorization") != "" {
 		return tst.base.RoundTrip(req)
 	}
-	return tst.ort.RoundTrip(req)
+	resp, err := tst.ort.RoundTrip(req)
+	if err == nil && resp != nil && resp.StatusCode == 401 && tst.src != nil {
+		tst.src.ResetToken()
+		return tst.ort.RoundTrip(req)
```

Comment:
Replaying the request is not safe for requests with a body. The simplest thing to do here is to just reset the token and return the error.

Reply:
if the request fails at authorization layer, replaying would not trigger any system state change right?

yea, return the error probably simpler and let the client to retry.